app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt

```diff
@@ -1,170 +1,267 @@
+            }
+            val addBookTitle = binding.titleET.text.toString()
+            val addBookWriter = binding.authorET.text.toString()
+            val addGenre = binding.genreET.text.toString()
+            val addCity = binding.cityET.text.toString()
+            val addContactMethod = binding.contactET.text.toString()
+
+            val someBooks = Adverts(
+                title = addBookTitle,
+                author = addBookWriter,
+                city = addCity,
                 genre = addGenre,
-                contact = addKontaktsatt,
+                contact = addContactMethod
             )
 
-            someBooks.adcreator = Firebase.auth.currentUser!!.uid
+            Firebase.auth.currentUser?.let { currentUser ->
+                someBooks.adCreator = currentUser.uid
+            }
 
-            if (currentannons == null) {
-                var adsave = myRef.push()
-                adsave.setValue(someBooks).addOnSuccessListener {
-                    uploadimage(adsave.key!!)
+            viewModel.advertSaveStatus.observe(viewLifecycleOwner) { isSuccessful ->
+                if (isSuccessful) {
+                    lifecycleScope.launch {
+                        // If there's an image, upload it.
+                        adImage?.let {
+                            // Use the adId for the image upload
+                            val adId = someBooks.adId ?: return@launch
+                            uploadImage(adId)
+
+                            // Introduce a delay of 3 seconds (or however long you want)
+                            delay(3000)
+                        }
+
+                        // After the delay, navigate back
+                        navigateBack()
+                    }
+                } else {
+                    Toast.makeText(
+                        requireContext(),
+                        "Failed to save advert. Please try again.",
+                        Toast.LENGTH_SHORT
+                    ).show()
                 }
+            }
+
+            if (currentAd == null) {
+                viewModel.saveAdvert(someBooks)
             } else {
-                myRef.child(currentannons!!.adid).setValue(someBooks).addOnSuccessListener {
-                    uploadimage(currentannons!!.adid)
+                currentAd?.adId?.let { existingAdId ->
+                    someBooks.adId = existingAdId
+                    viewModel.saveAdvert(someBooks)
                 }
             }
+        }
 
-            // val action = SkapaAnnonsFragmentDirections.actionSkapaAnnonsFragmentToAnnonsFragment()
-            // findNavController().navigate(R.id.action_skapaAnnonsFragment_to_annonsFragment)
-            findNavController().popBackStack()
+        viewModel.advertSaveStatus.observe(viewLifecycleOwner) { isSuccessful ->
+            if (isSuccessful) {
+                Toast.makeText(requireContext(), "Advert saved successfully!", Toast.LENGTH_SHORT)
```

Comment:
Hardcoded string

Reply:
Something is totally weird with whole PR. My latest changes are not these ones